src/Components/NotificationsProvider.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useMemo, useState } from "react";
+import NotificationsContext from "./NotificationsContext";
+import {
+  collection,
+  doc,
+  getCountFromServer,
+  getDocs,
+  limit,
+  onSnapshot,
+  orderBy,
+  query,
+  where,
+} from "firebase/firestore";
+import { auth, db } from "./Firebase";
+import { useAuthState } from "react-firebase-hooks/auth";
+
+export default function NotificationsProvider(props) {
+  const [user, loading, error] = useAuthState(auth);
+  const [notifications, setNotifications] = useState();
+  const [showMore, setShowMore] = useState(false);
+  const [hideBadge, setHideBadge] = useState(true);
+  const [moreNotiRequests, setMoreNotiRequests] = useState(0);
+
+  // Read initial state from Firestore, or else use blank array.
+  // It calls Firestore one time, after user has been authed.
+  useMemo(() => {
+    if (!user) return;
+    // Get notifications
+    const notisRef = collection(
+      db,
+      "notifications",
+      user?.uid,
+      "usersNotifications"
+    );
+    const q = query(
+      notisRef,
+      orderBy("time", "desc"),
+      limit(5 + moreNotiRequests * 5)
+    );
+    onSnapshot(q, (querySnapshot) => {
```

Comment:
Thanks for clarifying! 

It's the detaching/attaching each time a user clicks "See More" (and `moreNotiRequests` increments) that give me pause here.

Seems like there's a better method out there that wouldn't re-read the already shown content...maybe a more true paginated approach that only listens to chunks of 5 documents at once...but I suppose this works for now!

Reply:
Yeah, I agree that closing and opening a listener isn't really ideal when the user wants to load more.  The way it was before, you would end up adding a new listener as well (you would need to since the query to listen to changed -- a larger size), but the old listener would stay open as well, which could cause listeners to stack up if there were many pages to load.  _Probably_ the right thing to do here is construct the notifications list via two sources: 1) a listener that keeps the first page up-to-date and loads new items when they come in, and 2) a pagination system that makes one time fetches of deeper pages of notifications when the user paginates.  Then the UI concatenates the results from these two to create the list of notifications the user sees.  This would prevent needing new listeners on pagination.